### PR TITLE
add default retry

### PR DESCRIPTION
**Pull request by dileepfrog**

### CODE REVIEW

**Review by kfang — COMMENTED** [review]
Actually, is it possible to add the default retry to the zuora.setup() call?

**Comment by dileepfrog** [comment]
Doesn't look like setup() makes any network calls https://github.com/HeadspaceMeditation/zuora-request-service/blob/main/index.js#L16

**Comment by kfang** [comment]
@dileepfrog , yeah, setup() initializes a bunch of default configs like the baseUrl and the keys that all the subsequent network calls use. 

I was thinking you could use the setup() to store a 'global' defaultOptions and then merge those in when you actually make the request to zuora.

**Review by kfang — DISMISSED** [review]

File: zuora-api/proxied-request.js [hunk 1]
```diff
@@ -14,9 +14,8 @@ const isObject = require('lodash/isObject');
  * @return {[Object]}                suitable object for making http requests from nice-request
  */
 const mergeOptions = (requestOptions, extraOptions) => {
-  const result = isObject(extraOptions) ? Object.assign(requestOptions, extraOptions) : requestOptions;
-
-  return result;
+  let result = Object.assign(config.defaultRequestOptions(), requestOptions);
```
Comment (kfang) [inline zuora-api/proxied-request.js]: won't Object.assign modify defaultRequestOptions with the requestOptions you pass in? Might be safer to use Object.assign({}, defaultRequestOptions, requestOptions).

File: gulpfile.js [hunk 2]
```diff
@@ -56,25 +65,32 @@ gulp.task('lint', ['enforce-quality'], () => {
...
+  return gulp.watch(
+    ['index.js', 'index.test.js', 'zuora-config.test.js', 'zuora-api/**'],
+    ['lint'],
+    event => {
+      if (event.type === 'deleted' && cache.caches.eslint) {
+        // remove deleted files from cache
+        delete cache.caches.eslint[event.path];
+      }
+    },
+  );
```
Comment (kfang) [inline gulpfile.js:77]: lol, that linter...

**Review by kfang — DISMISSED** [review]

**Review by kfang — APPROVED** [review]

**Pull request merged**